[2025-04-21T14:30:27.450+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-04-21T14:30:27.539+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: perbaikan_skripsi.predict_and_store_result scheduled__2025-04-21T14:25:00+00:00 [queued]>
[2025-04-21T14:30:27.631+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: perbaikan_skripsi.predict_and_store_result scheduled__2025-04-21T14:25:00+00:00 [queued]>
[2025-04-21T14:30:27.642+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 4
[2025-04-21T14:30:28.005+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): predict_and_store_result> on 2025-04-21 14:25:00+00:00
[2025-04-21T14:30:28.242+0000] {logging_mixin.py:190} WARNING - /home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=16118) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-04-21T14:30:28.244+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'perbaikan_skripsi', 'predict_and_store_result', 'scheduled__2025-04-21T14:25:00+00:00', '--job-id', '133', '--raw', '--subdir', 'DAGS_FOLDER/tugas_akhir.py', '--cfg-path', '/tmp/tmpjokk0f03']
[2025-04-21T14:30:28.256+0000] {standard_task_runner.py:105} INFO - Job 133: Subtask predict_and_store_result
[2025-04-21T14:30:28.254+0000] {standard_task_runner.py:72} INFO - Started process 16266 to run task
[2025-04-21T14:30:29.177+0000] {task_command.py:467} INFO - Running <TaskInstance: perbaikan_skripsi.predict_and_store_result scheduled__2025-04-21T14:25:00+00:00 [running]> on host 1daba33e39bb
[2025-04-21T14:30:30.774+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Dion Rikki' AIRFLOW_CTX_DAG_ID='perbaikan_skripsi' AIRFLOW_CTX_TASK_ID='predict_and_store_result' AIRFLOW_CTX_EXECUTION_DATE='2025-04-21T14:25:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-04-21T14:25:00+00:00'
[2025-04-21T14:30:30.787+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-04-21T14:30:30.788+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-04-21T14:30:30.791+0000] {logging_mixin.py:190} INFO - Current task name:predict_and_store_result state:running start_date:2025-04-21 14:30:27.546196+00:00
[2025-04-21T14:30:30.800+0000] {logging_mixin.py:190} INFO - Dag name:perbaikan_skripsi and current dag run status:running
[2025-04-21T14:30:30.819+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-04-21T14:30:30.824+0000] {logging_mixin.py:190} INFO - üîç [2] Memproses data dan melakukan prediksi...
[2025-04-21T14:30:31.171+0000] {logging_mixin.py:190} INFO - ‚úÖ Data siap diproses: 2 baris
   id  Temperature  Humidity  ...  nitrogen phosphorous potassium
0  38         25.3      83.0  ...       0.0         0.0       0.0
1  39         25.3      83.0  ...       0.0         0.0       0.0

[2 rows x 9 columns]
[2025-04-21T14:30:31.176+0000] {logging_mixin.py:190} INFO - üìù Menyesuaikan nama kolom dengan model training...
[2025-04-21T14:30:31.180+0000] {logging_mixin.py:190} INFO - üîß Melakukan encoding kategori Soil_Type dan Crop_Type...
[2025-04-21T14:30:31.212+0000] {logging_mixin.py:190} INFO - üß† Menyiapkan fitur untuk prediksi...
[2025-04-21T14:30:31.234+0000] {logging_mixin.py:190} INFO - ü§ñ Menjalankan model prediksi...
[2025-04-21T14:30:31.351+0000] {logging_mixin.py:190} INFO - ‚úÖ Prediksi selesai:
   id predicted_fertilizer
0  38             10-10-10
1  39             10-10-10
[2025-04-21T14:30:31.357+0000] {logging_mixin.py:190} INFO - üíæ Menyimpan hasil ke tabel fertilizer_predictions...
[2025-04-21T14:30:31.545+0000] {logging_mixin.py:190} INFO - ‚ùå Error saat menyimpan ke database: (pymysql.err.IntegrityError) (1062, "Duplicate entry '38' for key 'fertilizer_predictions.sensor_data_id'")
[SQL: INSERT INTO fertilizer_predictions (sensor_data_id, predicted_fertilizer, created_at) VALUES (%(sensor_data_id)s, %(predicted_fertilizer)s, %(created_at)s)]
[parameters: ({'sensor_data_id': 38, 'predicted_fertilizer': '10-10-10', 'created_at': datetime.datetime(2025, 4, 21, 14, 30, 31, 389139)}, {'sensor_data_id': 39, 'predicted_fertilizer': '10-10-10', 'created_at': datetime.datetime(2025, 4, 21, 14, 30, 31, 389139)})]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
[2025-04-21T14:30:31.547+0000] {python.py:240} INFO - Done. Returned value was: None
[2025-04-21T14:30:31.595+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-04-21T14:30:31.598+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=perbaikan_skripsi, task_id=predict_and_store_result, run_id=scheduled__2025-04-21T14:25:00+00:00, execution_date=20250421T142500, start_date=20250421T143027, end_date=20250421T143031
[2025-04-21T14:30:31.691+0000] {logging_mixin.py:190} INFO - Task instance in success state
[2025-04-21T14:30:31.693+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: running
[2025-04-21T14:30:31.696+0000] {logging_mixin.py:190} INFO - Dag name:perbaikan_skripsi queued_at:2025-04-21 14:26:01.020939+00:00
[2025-04-21T14:30:31.697+0000] {logging_mixin.py:190} INFO - Task hostname:1daba33e39bb operator:PythonOperator
[2025-04-21T14:30:31.752+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2025-04-21T14:30:31.825+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-04-21T14:30:31.829+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
